spark/v3.5/spark-runtime/NOTICE

```diff
@@ -497,12 +680,46 @@ file:
 
 --------------------------------------------------------------------------------
 
-This binary artifact includes Project Nessie with the following in its NOTICE
-file:
+This binary artifact includes Project Nessie 0.102.2 with the following in its NOTICE file:
 
 | Dremio
 | Copyright 2015-2017 Dremio Corporation
 |
 | This product includes software developed at
 | The Apache Software Foundation (http://www.apache.org/).
-
+|
+| Apache Avro
+| Copyright The Apache Software Foundation
+|
+| Apache Commons
+| Copyright The Apache Software Foundation
+|
+| Apache Curator
+| Copyright The Apache Software Foundation
+|
+| Apache Hadoop
+| Copyright The Apache Software Foundation
+|
+| Apache HTTP Components
+| Copyright The Apache Software Foundation
+|
+| Apache Iceberg
+| Copyright The Apache Software Foundation
+|
+| Apache Kerby
+| Copyright The Apache Software Foundation
+|
+| This product includes additional software licensed under the terms
+| of the following licenses, see LICENSE file:
+| * Apache Software License, Version 2.0
+| * Creative Commons 1.0 Universal
+| * Eclipse Distribution License, Version 1.0
+| * Eclipse Public License, Version 1.0
+| * Eclipse Public License, Version 2.0
+| * GNU General Public License, Version 2 with the GNU Classpath Exception
```

Comment:
This is a release blocker. If there is GPL code in Nessie, I think that we should remove the dependency from Iceberg. @jbonofre, @amogh-jahagirdar

Reply:
Taking a look. Does the fact that it's a "classpath exception" matter or not? https://softwareengineering.stackexchange.com/questions/119436/what-does-gpl-with-classpath-exception-mean-in-practice 

From my reading of this, the classpath exception has the effect of not having the typical incompatibility between Apache and GPL licenses but I could be oversimplifying/misunderstanding that.

Reply:
https://www.apache.org/legal/resolved.html#category-x

I think "GNU Classpath Exception" can be included in ASF project (under special exception) 

> Places restrictions on larger works:
[GNU GPL 1, 2, 3](http://www.opensource.org/licenses/gpl-license.php)
Special exceptions to the GNU GPL (e.g. GNU Classpath) unless otherwise permitted elsewhere on this page.

Reply:
Yes and no: Classpath Extension allows you to add GPL dependency in your classpath (that's the name). But you can not bundle GPL dependency in ASF project.